File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.java
class Solution {
    public int partitionArray(int[] nums, int k) {
        Arrays.sort(nums);

        int x = nums[0];
        int upperLimit = x + k;
        int res = 1;
        for(int i = 0; i < nums.length; i++){
            
            if(nums[i] > upperLimit){
                res++;
                x = nums[i];
                upperLimit = x +k;
            }
        }
        return res;
    }
}

File: 2387-partition-array-such-that-maximum-difference-is-k/2387-partition-array-such-that-maximum-difference-is-k.rs
impl Solution {
    pub fn partition_array(nums: Vec<i32>, k: i32) -> i32 {
        //sort the partition_array
        let mut nums = nums;
        nums.sort();
        let mut res = 1;
        let mut start = nums[0];
        for i in 1..nums.len(){
            if nums[i] - start > k{
                start = nums[i];
                res += 1;
            } 
        }
        res
    }
}